### ISSUE TITLE
Does not work "on my computer" :-)

### ISSUE THREAD

**Issue by sturlath** [filipw/Strathweb.Samples.CSharp.NoVisibilityChecks#1]
Hi I just downloaded the example after I read your blog post, after witch I tried to use the  [IgnoresAccessChecksTo] attribute without luck.

And after downloading your example and trying to build it I get the same access protection level errors.

Am I missing something here? Should't it just build or is there some part missing? Can't figure it out.

**This is my VS info if that helps**
Microsoft Visual Studio Professional 2017 
Version 15.8.2
VisualStudio.15.Release/15.8.2+28010.2016
Microsoft .NET Framework
Version 4.7.03056

**Comment by sturlath** [comment]
Ok...ok... I just run the Compiler project and that in affect runs the code in the CrazyProgram project even though I get the protection level access errors there. 

So I basically need the code in the [Compiler Main](https://github.com/filipw/Strathweb.Samples.CSharp.NoVisibilityChecks/blob/master/Compiler/Program.cs) to run the code and **not just** adding [[assembly: IgnoresAccessChecksTo("Calculator")]](https://github.com/filipw/Strathweb.Samples.CSharp.NoVisibilityChecks/blob/master/CrazyProgram/Program.cs)? I will have to see how I can use this code with my tests in a generic manner.

**I was hoping to be able to add this attribute to my unit test namespace and have access to all the private method**s. 

But anyway I need be able to **suppress these errors** if the solution is to build. **How do I do that?**

Again... am I missing something? :-)

![image](https://user-images.githubusercontent.com/2386572/49291275-a3cf8b80-f4a1-11e8-88bf-9c36c5411e35.png)

**Comment by sturlath** [comment]
Another update. I managed to use [this nuget package](https://github.com/aelij/IgnoresAccessChecksToGenerator/issues/5) to get rid of all the errors and access the internal classes but **not** the one on the private method. Thought you might find that package interesting

![image](https://user-images.githubusercontent.com/2386572/49291208-69fe8500-f4a1-11e8-93e0-b20073b073d4.png)

**Comment by sturlath** [comment]
Are we also just talking about giving access to internal methods and **not private**? Just like the [**InternalsVisibleToAttribute**] 

The [IgnoresAccessChecksToAttribute ](https://github.com/aelij/IgnoresAccessChecksToGenerator/issues/5#issuecomment-443172162) nuget only works with internal.

**Comment by filipw** [comment]
> Are we also just talking about giving access to internal methods and **not private**?

As mentioned in the post, the compiler also need `BinderFlags.IgnoreAccessibility` to be set to be able to work with privates.

**Comment by sturlath** [comment]
Yes I’m seeing it now thanks. But how about the build/parsing errors? Any change I can get rid of them?

**Comment by filipw** [comment]
> Yes I’m seeing it now thanks. But how about the build/parsing errors? Any change I can get rid of them?

you can get rid of them only if you modify the compiler that is powering the language service. This is really not possible (AFAIK) in Visual Studio because you can't load custom Roslyn build in there.

you can do that in OmniSharp/VS Code though, by making the same change as here https://github.com/filipw/Strathweb.Samples.CSharp.NoVisibilityChecks/blob/master/Compiler/Program.cs#L24-L28 over here https://github.com/OmniSharp/omnisharp-roslyn/blob/ba7909c19ccd6513cbbcbf60f39339c790fe75b6/src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfoExtensions.cs#L15-L50

**Comment by Denis535** [comment]
[Cometary](https://github.com/71/Cometary/blob/master/src/Cometary.Core/Hooks.cs#L215) customizes Roslyn via [DiagnosticAnalyzer](https://github.com/71/Cometary/blob/master/src/Cometary.Analyzer/CometaryAnalyzer.cs#L65) and [Ryder](https://github.com/71/Ryder).
May be it can be possible to change ```CSharpCompilationOptions``` with the same way?

But it would be perfect to inject via cecil into [Csc ](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Shared/Csc.cs#L16) constructor the next code:

    Arguments.CompilationOptions = Arguments.CompilationOptions.WithMetadataImportOptions(MetadataImportOptions.All).WithTopLevelBinderFlags(BinderFlags.IgnoreAccessibility);
